### PR TITLE
Add pull request template for PlaceCal

### LINKED ISSUE #2501: Add a PR template
## Description

We should prob have a PR template that prompts the author and reviewers to do some sanity checks including:

- [ ] Removing console.logs and prints and things
- [ ] Moving strings into locale files as we go
- [ ] Moving mountain_view components into view_components as we go
- [ ] Making sure code is in the right place  (e.g. separation of concerns)
- [ ] Making sure tests are in the right place
- [ ] Adding inline documentation to non-trivial methods
- [ ] A statement about what is tested for and what isn't and why
- [ ] Prompt to consider method and field naming before merge
- [ ] Prompt to consider refactoring and DRY considerations

## Motivation

Let's just do housekeeping as we go.

### PR DESCRIPTION
**Pull request by lenikadali**
# Checklist:

- [x] I have performed a self-review of my own code,
- [x] I have commented my code, particularly in hard-to-understand areas,
- [x] I have made corresponding changes to the documentation,
- [x] I have added tests that prove my fix is effective or that my feature works,
- [x] New and existing unit tests pass locally with my changes
- [x] Title include "WIP" if work is in progress.

Resolves #2501

### Motivation and Context

The PR aims to provide a pull request template that makes it easy for new contributors to the project to provide details regarding what their pull request does. It also helps reviewers easily and quickly understand what the pull request is meant to do and review it with the appropriate context in mind.

### Type of change

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

### How Has This Been Tested?

By viewing the resulting markdown file.

### How Will This Be Deployed?

N/A

### Screenshots

N/A

### CODE REVIEW

**Comment by kimadactyl** [comment]
Hey this is looking great!

There's a bunch of things noted down in the linked ticket - do you think any of them are worth pulling through?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+<!--- Provide a general summary of your changes in the Title above -->
+<!--Read comments, before committing pull request read checklist again
+
+# Checklist:
+
+- [ ] I have performed a self-review of my own code,
+- [ ] I have commented my code, particularly in hard-to-understand areas,
+- [ ] I have made corresponding changes to the documentation,
+- [ ] I have added tests that prove my fix is effective or that my feature works,
+- [ ] New and existing unit tests pass locally with my changes
```
Comment (kimadactyl) [inline .github/PULL_REQUEST_TEMPLATE.md:10]: I think this maybe isnt needed as its automated?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+- [ ] This change requires a documentation update
+- [ ] Documentation update
+
+### How Has This Been Tested?
+
+<!--- Please describe in detail how you tested your changes so that a reviewer can reproduce the results. -->
+<!--- Include details of your testing environment, and the tests you've run your self -->
+<!--- see how your change affects other areas of the code, etc. -->
+
+### How Will This Be Deployed?
```
Comment (kimadactyl) [inline .github/PULL_REQUEST_TEMPLATE.md:45]: i think in 99% of cases there will be no changes to deployment - maybe we could add this as a checkbox in type of change instead, i.e. "requires change to deployment proces"?

Comment (lenikadali) [reply .github/PULL_REQUEST_TEMPLATE.md:45]: I like this idea. Will do!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (lenikadali) [reply .github/PULL_REQUEST_TEMPLATE.md:10]: Sure. I think we could re-add it if it ever becomes an issue (often changes are committed, the tests are failing and there is a request for review)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (lenikadali) [reply .github/PULL_REQUEST_TEMPLATE.md:45]: Resolved in [453f671](https://github.com/geeksforsocialchange/PlaceCal/pull/2630/commits/453f671991dcc3d3ffd32332616652d07ad4d695)

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (lenikadali) [reply .github/PULL_REQUEST_TEMPLATE.md]: Resolved in [453f671](https://github.com/geeksforsocialchange/PlaceCal/pull/2630/commits/453f671991dcc3d3ffd32332616652d07ad4d695)

**Comment by kimadactyl** [comment]
> @kimadactyl this should be good to go now. Thanks for the review 🙌🏿

Did you have a look at the linked ticket and do you think those issues are either represented or not needed?

**Comment by lenikadali** [comment]
> > @kimadactyl this should be good to go now. Thanks for the review 🙌🏿
> 
> Did you have a look at the linked ticket and do you think those issues are either represented or not needed?

I think not needed. The other items could be moved into a separate guide on how to interact with pull requests, both for contributors and reviewers. That's how I've seen it done in other projects. Also can't confirm now but things like `console.log` statements should be covered out of the box by linters (ESLint, RuboCop etc.)

**Comment by kimadactyl** [comment]
Ugh #2556 is happening again, so annoying!

**Pull request merged**